src/map.rs

```diff
@@ -482,7 +482,7 @@ impl<K, V, S> HashMap<K, V, S> {
                 continue;
             }
 
-            if self.size_ctl.compare_and_swap(sc, -1, Ordering::SeqCst) == sc {
+            if self.size_ctl.compare_exchange(sc, -1, Ordering::SeqCst, Ordering::Relaxed) == Ok(sc) {
```

Comment:
This works, but is overly complex. With `compare_exchange` you should be able to use `.is_ok()` instead.